frontend/javascripts/oxalis/controller/combinations/tool_controls.ts

```diff
@@ -780,6 +784,171 @@ export class QuickSelectTool {
+      _delta: Point2,
+      pos: Point2,
+      plane: OrthoView | null | undefined,
+      evt: MouseEvent,
+    ) => {
+      // In case the tool was reset by the user, abort measuring.
+      if (lineMeasurementGeometry.wasReset && isMeasuring) {
+        isMeasuring = false;
+      }
+      if (plane !== initialPlane || !isMeasuring) {
+        MoveHandlers.moveWhenAltIsPressed(_delta, pos, plane, evt);
+        return;
+      }
+      const state = Store.getState();
+      const newPos = V3.floor(calculateGlobalPos(state, pos, initialPlane));
+      lineMeasurementGeometry.setTopPoint(newPos);
+      Store.dispatch(showMeasurementTooltipAction([evt.clientX, evt.clientY]));
+    };
+    const rightClick = (pos: Point2, plane: OrthoView, event: MouseEvent) => {
+      // In case the tool was reset by the user, abort measuring.
+      if (lineMeasurementGeometry.wasReset && isMeasuring) {
+        isMeasuring = false;
+        return;
+      }
+      if (isMeasuring) {
+        // Set the last point of the measurement and stop measuring.
+        mouseMove({ x: 0, y: 0 }, pos, plane, event);
+        isMeasuring = false;
+      } else {
+        // If the tool already stopped measuring, reset the tool.
+        lineMeasurementGeometry.reset();
+        lineMeasurementGeometry.hide();
+        Store.dispatch(hideMeasurementTooltipAction());
+      }
+    };
+    const leftClick = (pos: Point2, plane: OrthoView, event: MouseEvent) => {
+      // In case the tool was reset by the user, abort measuring.
+      if (lineMeasurementGeometry.wasReset && isMeasuring) {
+        isMeasuring = false;
+        return;
+      }
+      const currentTime = Date.now();
+      if (currentTime - lastLeftClickTime <= this.DOUBLE_CLICK_TIME_THRESHOLD) {
+        // A double click should also terminate measuring.
+        rightClick(pos, plane, event);
+        return;
+      }
+      lastLeftClickTime = currentTime;
+      // Set a new measurement point.
+      const state = Store.getState();
+      const position = V3.floor(calculateGlobalPos(state, pos, plane));
+      initialPlane = plane;
+      if (!isMeasuring) {
+        lineMeasurementGeometry.setStartPoint(position, plane);
+        isMeasuring = true;
+      } else {
+        lineMeasurementGeometry.addPoint(position);
+        Store.dispatch(showMeasurementTooltipAction([event.clientX, event.clientY]));
+      }
+    };
+    return {
+      mouseMove,
+      rightClick,
+      leftClick,
+    };
+  }
+
+  static getActionDescriptors(
+    _activeTool: AnnotationTool,
+    _useLegacyBindings: boolean,
+    _shiftKey: boolean,
+    _ctrlKey: boolean,
+    _altKey: boolean,
+  ): ActionDescriptor {
+    return {
+      leftClick: "Left Click to measure distance",
+      rightClick: "Finish Measurement and reset",
+    };
+  }
+
+  static onToolDeselected() {
+    const { lineMeasurementGeometry } = getSceneController();
+    lineMeasurementGeometry.reset();
+    lineMeasurementGeometry.hide();
```

Comment:
There are several places where reset and hide is called. How about adding a `resetAndHide` helper method?